runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolLogReplication.java

```diff
@@ -163,10 +163,10 @@ public static LogReplicationEntryMsg overrideMetadata(LogReplicationEntryMsg mes
     }
 
     public static ResponseMsg getLeadershipResponse(
-            HeaderMsg header, boolean isLeader, String nodeId) {
+            HeaderMsg header, boolean isLeader, String nodeId, boolean isStandby) {
         LogReplication.LogReplicationLeadershipResponseMsg request = LogReplication.LogReplicationLeadershipResponseMsg
                 .newBuilder()
-                .setIsLeader(isLeader)
+                .setIsLeader(isLeader && isStandby)
```

Comment:
dont you think this approach of calculating isLeader is a bit confusing?  Can we return something like an invalid code if the active receives a metadata request?

Reply:
I think the important thing here is to keep compatibility with older versions, so introducing a new message might not be our best approach.